### PR TITLE
[CAS-1169] Feature/add huawei push provider

### PR DESCRIPTION
**Pull request by JcMinarro**
**Depend on #2155** 

### 🎯 Goal
Add implementation to use Huawei Push Kit

### 🛠 Implementation details
Create a new Artifact that provides all needed implementation to use Huawei Push Kit with Stream.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -17,6 +17,7 @@ buildscript {
         maven { url "https://plugins.gradle.org/m2/" }
         google()
         mavenCentral()
+        maven { url 'https://developer.huawei.com/repo/' }
```
Comment (zsmb13) [inline build.gradle:20]: Can we please limit this to only the artifacts that we actually need to pull from this repo?

File: docusaurus/docs/Android/02-client/06-guides/01-push-notifications/01-overview.mdx [hunk 2]
```diff
@@ -18,7 +18,7 @@ To receive push notifications from Stream Chat, you'll need to:
 We support the following providers:
 
 - [Firebase](./02-firebase.mdx)
-- Huawei Push Kit (Coming soon, stay tuned!)
+- [Huawei Push Kit](./03-huawei.mdx)
```
Comment (zsmb13) [inline docusaurus/docs/Android/02-client/06-guides/01-push-notifications/01-overview.mdx:21]: Leaving this on a random line: we should update the changelog to mention this new module

File: stream-chat-android-pushprovider-huawei/build.gradle [hunk 3]
```diff
@@ -0,0 +1,89 @@
+import io.getstream.chat.android.Dependencies
+import io.getstream.chat.android.Configuration
+
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-kapt'
```
Comment (zsmb13) [inline stream-chat-android-pushprovider-huawei/build.gradle:6]: Let's not add `kapt` if we're not using it

File: stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiMessagingDelegate.kt [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            Device(
+                token = token,
+                pushProvider = PushProvider.HUAWEI,
+            )
+        )
+    }
+}
+
+@Throws(IllegalStateException::class)
+private fun RemoteMessage.toPushMessage() = when (isValid()) {
```
Comment (zsmb13) [inline stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiMessagingDelegate.kt:56]: We're calling this when the `RemoteMessage` has already been validated, we don't really need this extra validation. We could update this file to match how we do it in https://github.com/GetStream/stream-chat-android/blob/develop/stream-chat-android-pushprovider-firebase/src/main/java/io/getstream/chat/android/pushprovider/firebase/FirebaseMessagingDelegate.kt

File: stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiPushDeviceGenerator.kt [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    private val hmsInstanceId: HmsInstanceId = HmsInstanceId.getInstance(context)
+    private val logger = ChatLogger.get("ChatNotifications")
+
+    override fun isValidForThisDevice(context: Context): Boolean =
+        (
+            HuaweiApiAvailability.getInstance()
+                .isHuaweiMobileServicesAvailable(context) == ConnectionResult.SUCCESS
+            ).also {
+            logger.logI("Is Huawei available on on this device -> $it")
+        }
```
Comment (zsmb13) [inline stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiPushDeviceGenerator.kt:26]: Ouch, this formatting 😅 

Maybe we should just give the function a regular body?

File: stream-chat-android-ui-components-sample/src/debug/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt [hunk 6]
```diff
@@ -12,6 +12,9 @@ import io.getstream.chat.android.client.utils.internal.toggle.ToggleService
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
 
 object ApplicationConfigurator {
+
+    val HUAWEI_APP_ID = "104598359"
```
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/src/debug/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt:16]:
```suggestion
    const val HUAWEI_APP_ID = "104598359"
```

File: stream-chat-android-ui-components-sample/src/release/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt [hunk 7]
```diff
@@ -3,5 +3,6 @@ package io.getstream.chat.ui.sample.application
 import android.app.Application
 
 object ApplicationConfigurator {
+    val HUAWEI_APP_ID = "104598369"
```
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/src/release/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt:6]:
```suggestion
    const val HUAWEI_APP_ID = "104598369"
```

**Review by zsmb13 — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (JcMinarro) [reply build.gradle:20]: Ok, In need to add it to two modules, the one that implement Huawei pushes and our sample app

File: stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiMessagingDelegate.kt (hunk 4 above)
Comment (JcMinarro) [reply stream-chat-android-pushprovider-huawei/src/main/java/io/getstream/chat/android/pushprovider/huawei/HuaweiMessagingDelegate.kt:56]: :+1:

File: build.gradle (hunk 1 above)
Comment (JcMinarro) [reply build.gradle:20]: It is needed to apply the huawei plugin :(

File: docusaurus/docs/Android/02-client/06-guides/01-push-notifications/01-overview.mdx (hunk 2 above)
Comment (JcMinarro) [reply docusaurus/docs/Android/02-client/06-guides/01-push-notifications/01-overview.mdx:21]: DONE ;)

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**